odbc/src/api/functions.rs

```diff
@@ -1655,6 +1701,30 @@ mod util {
     use odbc_sys::{Integer, SmallInt, SqlReturn, WChar};
     use std::{cmp::min, ptr::copy_nonoverlapping};
 
+    /// input_wtext_to_string converts an input cstring to a rust String.
+    /// It assumes nul termination if the supplied length is negative.
+    #[allow(clippy::uninit_vec)]
```

Comment:
do we need to just allow the lint here? A quick google suggests that perhaps there's another solution by using `MaybeUninit`?

Reply:
I was looking at `MaybeUninit` and couldn't figure out how to get it to work with `copy_non_overlapping`. It told me I couldn't use the pointer to that vec in `copy_non_overlapping`, and a quick look didn't reveal any examples working with both `copy_non_overlapping` and `MaybeUninit`. I can definitely look again.

Anyway, this PR is definitely still very much a work in progress :)